activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -491,9 +491,10 @@ def execute_grouped_calculation(operation, column_name, distinct) # :nodoc:
           end
 
           key_types = group_columns.each_with_object({}) do |(aliaz, col_name), types|
-            types[aliaz] = type_for(col_name) do
-              calculated_data.column_types.fetch(aliaz, Type.default_value)
-            end
+            types[aliaz] = col_name.try(:type_caster) ||
```

Comment:
Hum, I prefer to avoid `.try` as it's a huge footgun.

I'm not sure I fully understand the patch here. Is `col_name` expected to be `nil` or to be a type that doesn't implement `#type_caster`?

Reply:
> Is col_name expected to be nil or to be a type that doesn't implement #type_caster?

This. It could be a string.

Reply:
Ok. Can we use a `case / when String / else` instead?

Reply:
Ah wait, I missed the:

>  (column.try(:type_caster)) is used in many other places through the calculations.rb file.

Let's keep it consistent we can refactor later. SOrry.